test/unit/codegen/codegen_llvm_ir.cpp

```diff
@@ -901,6 +901,61 @@ SCENARIO("Scalar state kernel", "[visitor][llvm]") {
     }
 }
 
+//=============================================================================
+// Gather for vectorised kernel
+//=============================================================================
+
+SCENARIO("Vectorised simple kernel", "[visitor][llvm]") {
+    GIVEN("An indirect indexing of voltage") {
+        std::string nmodl_text = R"(
+            NEURON {
+                SUFFIX hh
+                NONSPECIFIC_CURRENT i
+            }
+
+            STATE {}
+
+            ASSIGNED {
+                v (mV)
+            }
+
+            BREAKPOINT {
+                SOLVE states METHOD cnexp
+                i = 2
+            }
+
+            DERIVATIVE states {}
```

Comment:
Just a note : in future PR, we could add `var_x = v` in the DERIVATIVE block and that could be used as a test for gather.